### PR TITLE
Crafting Module

### PR DESCRIPTION
**Pull request by bbumblebee**
I know you have always rejected requests concerning Crafting Modules, but I didn't really like Placing huge CraftingPipe-complexes next to chest just to get Smelting on demand working. Since I didn't see any technical problems coming with it, I decided to write it myself. I'm using it now without any problems, and think it should be included, since it's a major improvement concerning Crafting/Smelting on demand. Btw, theres no recipe and icon yet.

### CODE REVIEW

**Comment by MrTJP** [comment]
Why were  entire files  deleted then added? it makes it hard to see where edits are.

**Comment by bbumblebee** [comment]
I'm sorry, that wasn't intended but somehow happened I didn't know what to do then. But if you want I might just redo the whole thing.

**Comment by bbumblebee** [comment]
Sorry that was a misclick.

**Comment by MrTJP** [comment]
before you redo it, you may want to talk to the guys on IRC to see if they would even consider pulling it. It would be a shame to do all that hard work for nothing…

esper.net
#rs485

**Comment by davboecki** [comment]
I still don't like that idea. It is not that I don't know how to implement that or that I don't want to invest the time. It's more that I don't want to have more than one crafting recipe in one pipe. Also there is no satelite module and I don't plan to add one either. Crafting is intended to be it's own pipe and I don't plan to change that. Also the reason, that you want to use it for more than one furnace recipe is something I don't like. If you want to use one machine for more than one purpose than you need several crafting pipes. But I strongly recommend to use one machine for one crafting recipe. Cause the ingrediance can stack. Otherwise a lot of stuff will bounce and cause lag.

**Comment by MrTJP** [comment]
not to go against dav or anything, but if you REALLY want to, you COULD make a "add-on" for yourself.  :)

**Comment by bbumblebee** [comment]
Well I personnally think that it would be an awful waste of recources to use one furnace for each recipe (esp. when using ic2 with electric/induction furnace). Atm (with one maching/furnace) you have to make a huge chest system which takes a lot of space and is reaally slow. I simply don't really see the disadvantage of a module (You should be able to fix bouncing stuff by buffering the satellite input, e.g. BC hopper...). What would be the use of a satellite module?

**Comment by bbumblebee** [comment]
> not to go against dav or anything, but if you REALLY want to, you COULD make a "add-on" for yourself. :)

I already thought about that, but you have to make some changes to the RequestManager, which is static and therefore quite difficult to overwrite from an external mod.

File: common/logisticspipes/modules/ModuleCrafting.java [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+	@Override
+	public void registerPosition(int xCoord, int yCoord, int zCoord, int slot) {
+		this.xCoord = xCoord;
+		this.yCoord = yCoord;
+		this.zCoord = zCoord;
+		this.slot = slot;
+	}
+
+	@Override
+	public SinkReply sinksItem(ItemStack item, int bestPriority,
```
Comment (davboecki) [inline common/logisticspipes/modules/ModuleCrafting.java:123]: Why do you sink to a crafting module ?

File: common/logisticspipes/LogisticsPipes.java [hunk 2]
```diff
@@ -144,7 +144,7 @@
 	//Log Requests
 	public static boolean DisplayRequests;
 
-	public static boolean DEBUG = "%DEBUG%".equals("%" + "DEBUG" + "%") || "%DEBUG%".equals("true");
+	public static boolean DEBUG = "%DEBUG%".equals("%" + "DEBUG" + "%") || "%DEBUG%".equals("true") || true;
```
Comment (davboecki) [inline common/logisticspipes/LogisticsPipes.java]: Don't do that inside a Pull request.

File: common/logisticspipes/gui/hud/modules/HUDCraftingModule.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+			BasicGuiHelper.renderItemIdentifierStackListIntoGui(module.displayList, null, 0, 13, 3, 1, 1, 18, 18, mc, true, true, true, true);
+		} else {
+			BasicGuiHelper.renderItemIdentifierStackListIntoGui(list, null, 0, -9, 0, 1, 1, 18, 18, mc, true, true, true, true);
+		}
+
+	}
+
+	@Override
+	public List<IHUDButton> getButtons() {
+		// TODO Auto-generated method stub
```
Comment (davboecki) [inline common/logisticspipes/gui/hud/modules/HUDCraftingModule.java]: Remove that TODO if this function should return null.

**Comment by davboecki** [comment]
I'm still struggling with this PR. I was thinking of allowing this as an High Tech item at the endgame level using diamond chipsets but i'm still not sure wether i should allow this.
By the way could you format your code so that there are not tabs used in one line and spaces inside the next. Please always use Tabs.

File: common/logisticspipes/modules/ModuleCrafting.java (hunk 1 above)
Comment (bbumblebee) [inline common/logisticspipes/modules/ModuleCrafting.java:123]: Well, you have to make the chassi accept the requested Items, and that was the easiest way. If you remove that the Chassi sends the Items back.

**Comment by bbumblebee** [comment]
> By the way could you format your code so that there are not tabs used in one line and spaces inside the next. Please always use Tabs.

Where exactly did I use spaces? Actually I do (try to) always use tabs, I could run an eclipse cleanup over the whole code but that would change a lot of lines.

File: common/logisticspipes/LogisticsPipes.java [hunk 4]
```diff
@@ -145,7 +145,7 @@
 	public static boolean DisplayRequests;
 
 	public static boolean DEBUG = "%DEBUG%".equals("%" + "DEBUG" + "%") || "%DEBUG%".equals("true");
-	public static String MCVersion = "%MCVERSION%";
+	public static String MCVersion = "1.4.7";
```
Comment (davboecki) [inline common/logisticspipes/LogisticsPipes.java:148]: Please also revert that.

**Comment by davboecki** [comment]
and could you try to rebase this to the current dev version.

**Comment by keplersj** [comment]
Wait? So this is happening? If this does happen, wouldn't it make sense to include a mini-hopper or two in module so that davboecki's fears of lag and bounce back don't come true (typically ;) ) I think that this should be an extremely end-game item, because of it's power-ability and and much it has the ability to condense. I think that i should buffer quite a bit of items and use a lot of LP. This should ease some fears of OP and lag.

**Comment by MrTJP** [comment]
Since originally this idea was vetoed, it would make more sense for one of the actual devs to implement this themselves. It doesn't seem like the most efficient way.

**Comment by bbumblebee** [comment]
> and could you try to rebase this to the current dev version.
> Yep, doing this on a new PR

**Pull request closed without merging**